ltr/concourse/train.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+import os
+import sagemaker
+import sys
+
+from datetime import date
+
+govuk_environment = os.environ["GOVUK_ENVIRONMENT"]
+image = os.environ["IMAGE"]
+
+s3_bucket = os.getenv("S3_BUCKET", f"govuk-{govuk_environment}-search-relevancy")
+instance_count = os.getenv("INSTANCE_COUNT", 1)
+instance_size = os.getenv("INSTANCE_SIZE", "ml.m5.large")
+
+role = sagemaker.get_execution_role()
+
+session = sagemaker.Session()
+
+# find latest training/test/validate data
+all_training_data = session.boto_session.client("s3").list_objects_v2(
+    Bucket=s3_bucket, Prefix="data/",
+)
+
+train_key = None
+test_key = None
+validate_key = None
+for key in sorted([obj["Key"] for obj in all_training_data["Contents"]], reverse=True):
+    if "train.txt" in key and train_key is None:
+        train_key = key
+    elif "test.txt" in key and test_key is None:
+        test_key = key
+    elif "validate.txt" in key and validate_key is None:
+        validate_key = key
+    if train_key is not None and validate_key is not None and test_key is not None:
+        break
+
+err = False
+if train_key is None:
+    print("Could not find train.txt")
+    err = True
+if test_key is None:
+    print("Could not find test.txt")
+    err = True
+if validate_key is None:
+    print("Could not find validate.txt")
+    err = True
+
+if err:
+    sys.exit(1)
+
+# train model
+estimator = sagemaker.estimator.Estimator(
+    image,
+    role,
+    instance_count,
+    instance_size,
+    output_path=f"s3://{s3_bucket}/model/{date.today().isoformat()}",
+    sagemaker_session=session,
+)
+
+estimator.fit(
+    inputs={
+        "train": f"s3://{s3_bucket}/{train_key}",
+        "test": f"s3://{s3_bucket}/{test_key}",
+        "validate": f"s3://{s3_bucket}/{validate_key}",
+    }
+)
```

Comment:
Something to fix: it seems that this pushes a model artefact to s3 even if the training fails - we don't want that.

Also, the model isn't stored under the name I expected, it's stored as (eg) `s3://govuk-integration-search-relevancy/model/2020-01-09/search-2020-01-09-16-06-21-491/output/model.tar.gz`.  Need to see if there's a way to exactly define the name.

Reply:
Actually it seems it doesn't write to s3 if the training fails, I was misinterpreting the output.